### PR TITLE
Move filters Button in Call Hierarchy view to the top level Bar

### PR DESCRIPTION
**Pull request by jannisCode**
## What it does
Before the changes the filters could only be accessed through the "View Menu" (three dots), which is inconvenient considering that the filters button is rather important and frequently used.
After the changes the filters can be accessed by clicking on the icon in the top level bar as well as the old way through the view menu.

## Before
image
## After
Ticket1_bearbeitet

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

File: org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/CallHierarchyViewPart.java
Comment (fedejeanne) [inline org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/CallHierarchyViewPart.java:1053]: Remove empty line

**Review by fedejeanne — CHANGES_REQUESTED** [review]
@jannisCode thank you for the contribution, it looks good and works as expected!

I have some minor change requests though, mostly regarding new lines but the important one is about extracting the internal class `org.eclipse.jdt.internal.ui.callhierarchy.CallHierarchyFiltersActionGroup.ShowFilterDialogAction` in order to reduce code duplication.

File: org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/CallHierarchyFiltersActionGroup.java
Comment (fedejeanne) [inline org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/CallHierarchyFiltersActionGroup.java:94]: No need to remove this line

File: org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/ShowFilterDialogAction.java
Comment (fedejeanne) [inline org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/ShowFilterDialogAction.java:36]: Remove this

File: org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/ShowFilterDialogAction.java
Comment (fedejeanne) [inline org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/ShowFilterDialogAction.java:30]: The tooltip should only be added to the button in the bar, not to the menu. Add a new parameter called `tooltipText` and pass `null` as a value when instantiating this action from the menu,

File: org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/ShowFilterDialogAction.java
Comment (fedejeanne) [inline org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/ShowFilterDialogAction.java:24]: On a second thought, rename this one to `ShowCallHierarchyFilterDialogAction`. The reason is that by moving this class out of its previous container class (`CallHierarchyFiltersActionGroup`) the name of the action is too generic (if you look for the **type** (`Ctrl` + `Shift` + `t`) `ShowFilterDialogAction` then you'll see that there are other 2 internal classes with that name.

**Review by fedejeanne — CHANGES_REQUESTED** [review]

**Review by Wittmaxi — APPROVED** [review]

**Comment by fedejeanne** [comment]
@jannisCode regarding this build error:
```
10:45:23  [ERROR] Failed to execute goal org.eclipse.tycho:tycho-packaging-plugin:4.0.9:validate-version (default-validate-version) on project org.eclipse.jdt.ui: Unqualified OSGi version 3.33.200.qualifier must match unqualified Maven version 3.33.100-SNAPSHOT for SNAPSHOT builds -> [Help 1]
10:45:23  org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho:tycho-packaging-plugin:4.0.9:validate-version (default-validate-version) on project org.eclipse.jdt.ui: Unqualified OSGi version 3.33.200.qualifier must match unqualified Maven version 3.33.100-SNAPSHOT for SNAPSHOT builds
```

It seems that the `pom.xml` of `org.eclipse.jdt.ui` also has to be updated (increase the version too). Please do it as part of the same commit were you bumped the `MANIFEST.MF` (amend).

**Review by fedejeanne — APPROVED** [review]

**Comment by fedejeanne** [comment]
Thank you for addressing my change request @jannisCode . The PR looks good for me but since I'm not a committer, I can't merge it. 

Maybe some committer wants to take a look and let the workflows run before reviewing/merging?

**Comment by vogella** [comment]
@fedejeanne I'm not a JDT committer

**Comment by jjohnstn** [comment]
@fedejeanne I'll look at it today.

**Review by jjohnstn — APPROVED** [review]
Looks good (all of @fedejeanne comments have been addressed) and works fine.

**Comment by jjohnstn** [comment]
Hi @jannisCode Thank you for the patch.

**Comment by merks** [comment]
@jjohnstn  @jannisCode 

After a pull, my IDE now has this error:

image

Indeed it was incremented 4 weeks ago:

image

I don't think we can undo the damage after this has been pulled into an I-Build.

**Comment by fedejeanne** [comment]
@merks thank you for reporting it. Do you know why was this not reported by the (GH) builds?

I noticed the same error locally when trying to fix the build in this PR but I thought it might be a false positive in my IDE. Increasing the version fixed the PR-builds. More details in my discussion (I'd like to continue the discussion there): https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/discussions/2406

**Pull request merged**
